### ISSUE TITLE
[1.16.4 G6] Not compatible with latest Forge

### ISSUE THREAD

**Issue by james090500** [sp614x/optifine#5242]
## Description of Issue
Using Optifine 1.16.4 HD_U_G6 on Forge will cause EntityJoinWorldEvent not to fire when a PlayerEntity joins the game. Removing OptiFine and everything works fine.

## Steps to Reproduce
Using forge 1.16.4 35.1.37. With this bit of magic code
```
@SubscribeEvent
public void onPlayerJoin(EntityJoinWorldEvent event) {
	System.out.println(event.getEntity().getClass().toString());
}
```

Joining a single player word, the game wont detect a PlayerEntity joining. Only a ServerPlayerEntity. However, Optifine 1.16.4 G5 and OptiFine 1.16.5 G6 work fine both detect a PlayerEntity joining. 

## OptiFine Version
Optifine 1.16.4 HD_U_G6 

## Installation Method
Forge

## Fabric/Forge Version
Forge 1.16.4 35.1.37

## Other Installed Mods
See Above

## Log Files/Crash Reports
[latest.log](https://github.com/sp614x/optifine/files/5852048/latest.log)
[debug.log](https://github.com/sp614x/optifine/files/5852049/debug.log)

## F3 Debug Screenshot
![image](https://user-images.githubusercontent.com/20372645/105410188-7b7aec00-5c29-11eb-990e-aa39331b78cf.png)

**Comment by james090500** [comment]
Doing some further checks because when I originally wrote this I realised my mistake. EntityJoinWorldEvent is firing for players, but not firing if the world isRemote();

For example here is the code snippet
```
@SubscribeEvent
public void onPlayerJoin(EntityJoinWorldEvent event) {
	System.out.println(event.getEntity().getClass().toString() + " " + (event.getEntity() instanceof PlayerEntity));
	if(event.getEntity() instanceof PlayerEntity) {
		System.out.println(event.getWorld().isRemote());
	}
}
```

With OptiFine: https://paste.gg/p/anonymous/576051c69c704e5aa35462a289aca4bc (line 170 - 173)
Without OptiFine: https://paste.gg/p/anonymous/e102a3b6908d4694ae397da26041b0fa (line 75 - 82)

**Comment by Shanghai-** [comment]
This is causing problems in a few mods (see [this mekanism issue](https://github.com/mekanism/Mekanism/issues/6892#issuecomment-761909341)), so a quick fix would be stellar.

**Comment by TechniumUnlimited** [comment]
Are we aware whether the later versions of optifine fixed this issue? I am still on my custom version of mekanism to sidestep the problem.. but they just updated and I'd like to go back to proper mekanism xD

**Comment by james090500** [comment]
> Are we aware whether the later versions of optifine fixed this issue? I am still on my custom version of mekanism to sidestep the problem.. but they just updated and I'd like to go back to proper mekanism xD

OptiFine 1.16.5 fixes the issue, there are no new version of 1.16.4

**Comment by Shanghai-** [comment]
Confirmed that the latest preview of Optifine 1.16.5 fixes this for me. (And it's compatible with MC 1.16.4)

**Comment by sp614x** [comment]
Fixed in 1.16.4 G7.
